src/irmin-pack/unix/gc.ml

```diff
@@ -586,6 +586,8 @@ module Make (Args : Args) = struct
                 let () = Lwt.wakeup_later t.resolver (Ok !s) in
                 Ok (`Finalised !s)
             | _ ->
+                t.stats <- Some !s;
+                Fm.cleanup ~root:t.root ~generation:(t.generation - 1);
                 let err = gc_errors status gc_output in
                 let () = Lwt.wakeup_later t.resolver err in
```

Comment:
`t.stats <- Some !s` is the quickfix to mark the GC as finished (otherwise it's illegal to call `wakeup_later` a second time)

Reply:
Hmm, I wonder if we should also set `t.running_gc` to `None` in the `Error` case in `ext.ml`'s `finalise_exn`? It sort of works since it would get cleaned up with the next call to `finalise_exn` (since you are setting stats here), but that doesn't seem like the best approach.
https://github.com/mirage/irmin/blob/410b778edc7e577c2eb7d2af7386f3040aa0f9f9/src/irmin-pack/unix/ext.ml#L297-L301

Reply:
Ha thanks I missed that! It's indeed a lot cleaner to unset the `running_gc` than the previous hack :)

Reply:
:pray: I think we probably need both -- `t.running_gc <- None` to signal no GC is running for the repo and `t.stats <- Some !s` to signal the gc process is finished. It would be nice to actually track the state of the GC process (instead of implicitly through stats option) but we can do that in a future PR.